api/reference/1.0/notes/index.md

```diff
@@ -0,0 +1,56 @@
+---
+title: Overview
+---
+
+# Notes
+
+`Notes` are textual annotations you can add to Launch resources of these types:
+- `DataElements`
+- `RuleComponents`
+- `Rules`
+- `Extensions`
+- `Libraries`
+- `Properties`
+
+Those six types are collectively known as the `Notable` resources.
+
+When a `Notable` resource is deleted, its associated `Notes` are also deleted.
+
+## `Notes` are Comments
+Launch doesn't attach any meaning to `Notes`.
+Adding a `Note` won't change the behavior of your `Property`.
+
+You might use `Notes` to:
+- provide background information
+- function as to-do lists
+- pass along resource usage advice
+- give instructions to other team members
+- record historical context
+- remind yourself why this resource is built the way it is
+
+But Launch ignores `Notes` so you are free to use them any way you want.
+
+## Revisable Notables
+
+In the case of revisable types
+(`DataElements`, `RuleComponents`, `Rules`, and `Libraries`),
+any `Notes` must be created on the current (`head`)
+revision. They may not be attached to other revisions.
+
+However, `Notes` may be _read_ from revisions. In such cases, the endpoints
+will return only the `Notes` that existed before the creation of the revision.
+They provide a snapshot of the annotations as they were when the revision was cut.
+In contrast, reading `Notes` from the current (`head`) revision will return all
+its `Notes`.
+
+## Limitations
+
+Each `Note` is limited to 512 Unicode characters.
+
+`Notes` are not editable.
```

Comment:
@thebenrob I realize this is feature parity with DTM, but I question if historical notes should be editable. As you build out an implementation, you might make changes to Rule A which depends on DataElement A. If you can't revise your notes on DataElement A, you now have to edit it and cut a revision. This seems unnecessary just to add some documentation.

Reply:
Revisions don't have notes, when you ask for a revision's notes you get those from head before created before the revision's creation. This means once a revision is cut, you cannot add notes to it, just to what the next revision will be. You also cannot add notes to a revision, like `This is the revision where super bad bug was introduced`.

Head can have any number of notes, so you don't need to cut a new revision per se.
The way we've approximated notes for revisions is really weird to me. As a user I can't necessarily do the things I would like to do.  If we added notes to revisions, then displaying them becomes problematic because you have 2 time axis. One for revision creation date and one for note creation date. I'd like to see all of the notes for some origin, do I see the notes in time order and show revisions out of order? Do I show notes primarily in the order of the revision, and second by create date of the note?  
If I cut a revision of a rule, add the rule to the library. Then submit the library. The approver wants to say, `this is borked, yo! Change the implementation of x`. That note can only apply to head, which could be very different from the revision that was submitted.

Reply:
Yah, I think it might be a nice feature to make notes editable for a revision, but I don't think it's the end of the world not to have it. You can always add a new note to head that says something like `I broke purchase tracking in revision 3 and fixed it in revision 4`. 

Just wanted to bring this up in case we deem it a feature worth having.